### ISSUE TITLE
ItemBatch can sometimes have an undefined Item

### ISSUE THREAD

**Issue by joshxg** [openmsupply/mobile#2448]
## Error in mSupply Mobile

**TypeError** in **MainActivity**
undefined is not an object (evaluating 'o.realItem')

[View on Bugsnag](https://app.bugsnag.com/sustainable-solutions-nz-ltd/msupply-mobile/errors/5e5d92afd76ba6001939533e?event_id=5e5d92af005754ebd1ff0000&i=gh&m=ci)

## Stacktrace

    src/database/utilities/createRecord.js:282 - createStocktakeItem
    src/database/utilities/createRecord.js:474 - createRecord
    src/database/DataTypes/Stocktake.js:85 - 
    src/database/DataTypes/Stocktake.js:82 - value
    src/navigation/actions.js:326 - 
    src/database/UIDatabase.js:183 - value
    src/navigation/actions.js:323 - 
    src/pages/StocktakeManagePage.js:67 - 
    src/mSupplyMobileApp.js:141 - 

[View full stacktrace](https://app.bugsnag.com/sustainable-solutions-nz-ltd/msupply-mobile/errors/5e5d92afd76ba6001939533e?event_id=5e5d92af005754ebd1ff0000&i=gh&m=ci)

*Created automatically via Bugsnag*

**Comment by joshxg** [comment]
From the bugsnag before it 404's 🤣 

![image](https://user-images.githubusercontent.com/35858975/75728054-8ac02500-5d4b-11ea-8881-751233974f91.png)


--------

In `Stocktake.js`:

```
addBatchIfUnique(database, stocktakeBatch) {
    // TODO: rename method to addBatch.
    addBatchToParent(stocktakeBatch, this, () =>
      createRecord(database, 'StocktakeItem', this, stocktakeBatch.itemBatch.item)
    );
  }
```

Where `ItemBatch.item` is an optional field.

This is then destructured in `createRecord` : 

```
const createStocktakeItem = (database, stocktake, item) => {
  // Handle cross reference items.
  const { realItem } = item;

  const stocktakeItem = database.create('StocktakeItem', {
    id: generateUUID(),
    item: realItem,
    stocktake,
  });

  stocktake.items.push(stocktakeItem);
  database.save('Stocktake', stocktake);

  return stocktakeItem;
};
```



This hasn't been touched in yonks so it doesn't happen to often - still good to fix soon!

**Comment by wlthomson** [comment]
Fixed by #2448.

**Comment by wlthomson** [comment]
Difficult to replicate this one, but hasn't been snagged since the fix, so moving to done!
